gui/src/pages/Process.svelte

```diff
@@ -0,0 +1,213 @@
+  import * as router from 'svelte-spa-router';
+  import Layout from '../components/Layout.svelte';
+  import { api } from '../lib/api';
+  import { onDestroy, onMount } from 'svelte';
+  import { fetchProcesses, refreshAllProcesses, processes } from '../lib/process/store';
+  import type { RemoteData } from '../lib/api';
+  import type { ComponentDetails } from '../lib/process/types';
+  import sparkline from '@fnando/sparkline';
+  export let params = { workspace: '', process: '' };
+
+  const workspaceId = params.workspace;
+  const workspace = api.workspace(workspaceId);
+
+  const processId = params.process;
+
+  let processList: RemoteData<ComponentDetails[]> = { stage: 'pending' };
+  const unsubscribeProcesses = processes.subscribe((processes) => {
+    processList = processes;
+  });
+
+  let refreshInterval: any;
+  let process: ComponentDetails | null = null
+
+  const cpuPercentages: number[] = []
+
+  onMount(() => {
+    fetchProcesses(workspace);
+
+    // TODO: Server-sent events or websockets!
+    refreshInterval = setInterval(() => {
+      refreshAllProcesses(workspace);
+      if (processList.stage === "success") {
+        process = processList.data.filter(p => p.id === processId)[0];
+      }
+      if (process && process.status.running) {
+        cpuPercentages.push(process.status.CPUPercent)
+        if (cpuPercentages.length > 100) {
+          cpuPercentages.shift()
+        }
+        const sparklineSvg = document.querySelector(".sparkline");
+        if (cpuPercentages.some(p => p !== 0) && sparklineSvg && sparklineSvg instanceof SVGSVGElement) {
+          sparkline(
+            sparklineSvg,
+            cpuPercentages,
+            {interactive: true}
+          );
+        }
+      }
+    }, 1000);
+  });
+
+  onDestroy(() => {
+    clearInterval(refreshInterval);
+    unsubscribeProcesses();
+  });
+
+// From https://stackoverflow.com/a/14919494
+/**
+ * Format bytes as human-readable text.
+ * 
+ * @param bytes Number of bytes.
+ * @param si True to use metric (SI) units, aka powers of 1000. False to use 
+ *           binary (IEC), aka powers of 1024.
+ * @param dp Number of decimal places to display.
+ * 
+ * @return Formatted string.
+ */
+function humanFileSize(bytes: number, si=false, dp=1): string {
+  const thresh = si ? 1000 : 1024;
+
+  if (Math.abs(bytes) < thresh) {
+    return bytes + ' B';
+  }
+
+  const units = si 
+    ? ['kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'] 
+    : ['KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
+  let u = -1;
+  const r = 10**dp;
+
+  do {
+    bytes /= thresh;
+    ++u;
+  } while (Math.round(Math.abs(bytes) * r) / r >= thresh && u < units.length - 1);
+
+
+  return bytes.toFixed(dp) + ' ' + units[u];
+}
+</script>
+
+<Layout>
+  <section>
+    {#if process}
+      <div>
+        <div id="heading">
+          <button class="back-button" on:click={() => void router.push(`#/workspaces/${encodeURIComponent(workspaceId)}/`)}>
+            🠔 Back
```

Comment:
We have some mechanism for putting back links in the header, tho I think it might change as the layout overall changes. cc @jwmza for guidance.